### DISCUSSION TITLE
Open in Editor with direnv support

### DISCUSSION THREAD

**Discussion by MunsMan** [sxyazi/yazi#1083] [1. Q&A] [answered]
Hello everyone,

I've been working on optimizing my development workflow by leveraging the power of [direnv](https://direnv.net/) in combination with [Nix devshells](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-develop.html) to manage project dependencies more efficiently. This setup has significantly streamlined my process by automatically loading environment variables and dependencies specific to my project's directory.
However, I've encountered a challenge when integrating this setup with Yazi, especially when opening files in NeoVim. Ideally, I want direnv to be triggered and load the necessary environment when Yazi opens a file in NeoVim. This step is crucial for ensuring that all required dependencies for the Language Server Protocol (LSP) are available. Unfortunately, I haven't been successful in making this work as expected.

Currently, I see two options, loading the environment on *open neovim* or *entering the directory*, but one of my primary concerns is the potential slowdown of Yazi if direnv is loaded synchronously upon entering the directory. To mitigate this, I'm considering an asynchronous loading mechanism in the background, leveraging the fact that project dependencies are locally cached for faster load times. This approach should theoretically provide a seamless experience without compromising Yazi's performance. Otherwise, loading the environment before opening neovim maybe the easiest solution. 

I'm reaching out to the community for insights, suggestions, or any existing solutions that could help address this integration challenge. Additionally, if there's a specific area where I can contribute to making this possible, I'm more than willing to lend a hand. I just need a pointer on where to start.

Thank you for taking the time to read this. I'm looking forward to your feedback and any assistance you can provide.

**Comment by sxyazi** [comment]
Hi, I haven't used `direnv` before, but from a quick look, it seems to be used for initializing different environments for different directories. 

So I think you can call `direnv` to apply these environment settings after the `cd` event is triggered in Yazi. This can be done through DDS, and the whole process is asynchronous, so it won't block the execution of other tasks. Here's a piece of code for reference:

https://github.com/sxyazi/yazi/blob/5f21998665d78fe34e1e1440fc83214fed4fb1aa/yazi-plugin/preset/plugins/zoxide.lua#L38-L47

And the docs can be found at https://yazi-rs.github.io/docs/plugins/utils#ps

**Comment by MunsMan** [comment, marked as answer]
Thank you for your support!

I was able to solve my problem with the following configuration:

```toml
[opener]
edit = [
{ run = "direnv exec . ${EDITOR} $1", desc = "$EDITOR", block = true, for = "unix" },
...
]
```
